### ISSUE TITLE
Cycle detection in `discover`?

### ISSUE THREAD

**Issue by kain88-de** [datreant/datreant.core#66]
Is there a detection for symbolic link cycles in the discover function?

**Comment by dotsdl** [comment]
``datreant.core.discover`` doesn't follow symbolic links since it uses ``scandir.walk`` with ``followlinks=False``, so this shouldn't be an issue at all. Have you experienced otherwise?

**Comment by kain88-de** [comment]
I haven't tested it. I just read it in the paper and remembered that a colleague can't use pycharm because he has symbolic link cycles. 

But I can't find this information in the docs. I think it would be nice to document this behavior otherwise someone who uses symbolic links might be surprised he can't discover his Treants anymore.

**Comment by dotsdl** [comment]
@kain88-de agreed. In fact, we'll go one further and add a ``followlinks`` kwarg to ``discover`` that lets links be followed if desired. Not sure if ``scandir.walk`` does any loop detection on its own, but will check. Not sure if we should do loop detection ourselves if it doesn't, but probably should in the interests of being useful.

**Comment by kain88-de** [comment]
I don't know if scandir has any loop detection. I doubt it though. I also wouldn't say we should do our own. cyclic links are a problem are a problem for a lot of people and the cycle can be loooong. 

I like the option to follow symbolic links though. But a warning about cyclic links in the docs though first. Then we can still see if we get generate some evil cyclic links if we really want to.
